pandas/core/indexes/interval.py

```diff
@@ -92,6 +94,18 @@ def _get_interval_closed_bounds(interval):
     return left, right
 
 
+def _maybe_convert_platform_interval(data):
+    """
+    Try to do platform conversion, with special casing for IntervalIndex
+    """
+    if isinstance(data, (list, tuple)) and len(data) == 0:
+        # GH 19016
+        # empty lists/tuples get object dtype by default, but this is not
+        # prohibited for IntervalIndex, so coerce to integer instead
```

Comment:
The comment here refers to the existing behavior of constructing an `IntervalIndex` from an empty list:
```python
In [2]: pd.IntervalIndex([])
Out[2]:
IntervalIndex([]
              closed='right',
              dtype='interval[object]')
```
The changes in this PR caused this to raise, since the inferred subtype was object, which is now being disallowed.  Changed this to be instead be integer by default.  Don't have any strong reason for choosing integer, so can change.